### PR TITLE
Client sdk in typescript with transaction processing support

### PR DESCRIPTION
**Pull request by smithbk**
## Description

The node client SDK was written in typescript which is more robust and maintainable than simple node.js and will help the quality of the client SDK.  Typescript compiles to node.js, so it is completely node.js compatible.  It now supports transaction processing for deploy (dev mode only currently), invoke, and query.  The README.md has been reworked.  Two sets of tests (chain-tests.js and asset-mgmt.js have been added).  Reference documentation has been added as noted in README.md 
## Motivation and Context

The node.js client SDK is needed to enable users to write node.js apps to transact on the blockchain.

Fixes #
## How Has This Been Tested?

Two new suites of tests as described in README.md have been added.  See the fabric/sdk/node/test/unit directory.
## Checklist:

- [x] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff)
- [x] Either no new documentation is required by this change, OR I added new documentation
- [x] Either no new tests are required by this change, OR I added new tests
- [x] I have run [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports), [go vet](https://golang.org/cmd/vet/), and [golint](https://github.com/golang/lint). I have cleaned up all valid errors and warnings in code I have added or modified. These tools may generate false positives. Don't be worried about ignoring some errors or warnings. The goal is clean, consistent, and readable code.

Signed-off-by: Keith Smith bksmith@us.ibm.com

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -15,9 +15,9 @@ Fixes #
 ## Checklist:
 <!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
 <!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
-- [ ] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff)
```
Comment (corecode) [inline .github/PULL_REQUEST_TEMPLATE.md:18]: The checkboxes get mangled by this change.

**Comment by corecode** [comment]
Something in your pull request is wrong, and it includes commits that are already in the tree.  I suggest investigating what is going on before merging.

**Comment by christo4ferris** [comment]
+1 to what @corecode said... this PR needs to be cleaned up.

**Comment by smithbk** [comment]
Opened a new pull request from a clean branch client-SDK, so closing this pull request.

**Pull request closed without merging**